storage/database/db_manager.go

```diff
@@ -412,62 +442,76 @@ func (dbm *databaseManager) NewBatch(dbEntryType DBEntryType) Batch {
 	return dbm.getDatabase(dbEntryType).NewBatch()
 }
 
-func (dbm *databaseManager) getStateTrieMigrationInfo() (bool, string) {
+func (dbm *databaseManager) getDBDir(dbEntry DBEntryType) string {
+	miscDB := dbm.getDatabase(MiscDB)
+	enc, _ := miscDB.Get(databaseDirKey(uint64(dbEntry)))
+	return string(enc)
+}
+
+func (dbm *databaseManager) setDBDir(dbEntry DBEntryType, newDBDir string) error {
+	miscDB := dbm.getDatabase(MiscDB)
+	return miscDB.Put(databaseDirKey(uint64(dbEntry)), []byte(newDBDir))
+}
+
+func (dbm *databaseManager) getStateTrieMigrationInfo() bool {
 	miscDB := dbm.getDatabase(MiscDB)
 	enc, _ := miscDB.Get(migrationStatusKey)
 	if len(enc) != 8 {
-		return false, ""
+		return false
 	}
 	status := binary.BigEndian.Uint64(enc)
 	if status == 0 {
-		return false, ""
+		return false
 	}
-
-	enc, _ = miscDB.Get(databaseDirKey(uint64(StateTrieDB)))
-	return true, string(enc)
+	return true
 }
 
-func createStateTrieMigrationDB(dbc *DBConfig, newDBDir string) Database {
+func newStateTrieMigrationDB(dbc *DBConfig, blockNum uint64) (Database, string) {
 	oldDBConfig := dbc
-	newDBConfig := getDBEntryConfig(oldDBConfig, StateTrieDB)
-	newDBConfig.Dir = newDBDir
+	newDBConfig := getDBEntryConfig(oldDBConfig, StateTrieMigrationDB)
+	newDBConfig.Dir = newDBConfig.Dir + strconv.FormatUint(blockNum, 10)
 	var newDB Database
 	var err error
 	if newDBConfig.NumStateTriePartitions > 1 {
-		newDB, err = newPartitionedDB(newDBConfig, StateTrieDB, newDBConfig.NumStateTriePartitions)
+		newDB, err = newPartitionedDB(newDBConfig, StateTrieMigrationDB, newDBConfig.NumStateTriePartitions)
 	} else {
-		newDB, err = newDatabase(newDBConfig, StateTrieDB)
+		newDB, err = newDatabase(newDBConfig, StateTrieMigrationDB)
 	}
 	if err != nil {
 		logger.Crit("Failed to create a new database for state trie migration", "err", err)
 	}
 	logger.Info("Created a new database for state trie migration", "newStateTrieDB", newDBConfig.Dir)
-	return newDB
+	return newDB, newDBConfig.Dir
 }
 
-func (dbm *databaseManager) SetStateTrieMigrationDB(blockNum uint64) {
+func (dbm *databaseManager) SetStateTrieMigrationDB(blockNum uint64) error {
```

Comment:
`blockNum` should be bigger than last migration `blockNumber`. Do you think it is better to check before migration?

Reply:
After migration, for getting the last migration blocknumber, we need to parsing the DB_path.
We can add the condition as you told.
I think this feature is not necessary.
Let's discuss and if it is needed, reflect the feature on software requirement document 1st.